node/engine/interpreter/context.go

```diff
@@ -550,6 +550,20 @@ func (e *executionContext) getVariable(name string) (Value, error) {
 				return nil, engine.ErrInvalidTxCtx
 			}
 			return makeText(e.engineCtx.TxContext.Authenticator), nil
+		case "leader":
+			if e.engineCtx.InvalidTxCtx {
+				return nil, engine.ErrInvalidTxCtx
+			}
+
+			// Get current leader from the validators system
```

Comment:
Hey @MicBun, maybe we could not rely on a new `getCurrentLeader` code if this information is extracted from the same source as `@height`, i.e. `e.engineCtx.TxContext.BlockContext.Proposer`, which already exists

From reading this code, I believe it's bug-free, but it's better if we need no extra code

It should be safe, as blocks are always proposed by the leader of a block (as much as height is deterministic)

Or did you find any reason not to keep only that?

Reply:
but for sure there could be things here that I'm just not perceiving correctly yet